contracts/service-provider-directory/src/state.rs

```diff
@@ -0,0 +1,176 @@
+use cosmwasm_std::{Addr, Coin, StdResult, Storage, Uint128};
+use cw_storage_plus::{Item, Map};
+use serde::{Deserialize, Serialize};
+
+// Storage keys
+pub const CONFIG_KEY: &str = "config";
+pub const SERVICE_ID_COUNTER_KEY: &str = "sidc";
+pub const SERVICES_KEY: &str = "services";
+
+// Storage
+pub const CONFIG: Item<Config> = Item::new(CONFIG_KEY);
+pub const SERVICES: Map<ServiceId, Service> = Map::new(SERVICES_KEY);
+pub const SERVICE_ID_COUNTER: Item<ServiceId> = Item::new(SERVICE_ID_COUNTER_KEY);
+
+/// The directory of services are indexed by [`ServiceId`].
+pub type ServiceId = u64;
+
+/// The type of services provider supported
+#[derive(Serialize, Deserialize, Copy, Clone, PartialEq, Eq, Debug)]
+pub enum ServiceType {
+    NetworkRequester,
+}
+
+impl std::fmt::Display for ServiceType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let service_type = match self {
+            ServiceType::NetworkRequester => "network_requester",
+        };
+        write!(f, "{service_type}")
+    }
+}
+
+/// The types of addresses supported.
+#[derive(Serialize, Deserialize, Clone, PartialEq, Eq, Debug)]
+pub enum NymAddress {
+    /// String representation of a nym address, which is of the form
+    /// client_id.client_enc@gateway_id.
+    Address(String),
+    // For the future when we have a nym-dns contract
+    //Name(String),
+}
+
+impl NymAddress {
+    /// Create a new nym address.
+    pub fn new(address: &str) -> Self {
+        Self::Address(address.to_string())
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, Debug)]
+pub struct Service {
+    /// The address of the service.
+    pub nym_address: NymAddress,
+    /// The service type.
+    pub service_type: ServiceType,
+    /// Service owner.
+    pub owner: Addr,
+    /// Block height at which the service was added.
+    pub block_height: u64,
+    /// The deposit used to announce the service.
+    pub deposit: Coin,
+}
+
+#[cfg(test)]
+mod test {
```

Comment:
it feels weird to have both `mod test` and `mod tests` in the same file. Maybe you should combine them or move something elsewhere?